handle arbitrary arg list

kylewhite21 opened the pull request:
I have an Elixir app using this library, but was hitting an error when formatting a list of pids:
```
{:error, :badarg, [{:unicode, :characters_to_binary, [[#PID<0.2266.0>]]
```
This should handle that case

kylewhite21 committed:
handle arbitrary arg list

seriyps commented:
Hi, sorry for delay. Merged!
Would be cool to have a test for this too though.

kylewhite21 commented:
Hey @seriyps thanks for merging. I ran your pre commit checks (just now, not before the PR 🙈) and there are failing tests prior to my change. If you clear those up I can add tests for this.

seriyps commented:
The failing checks are due to code linter complaining about nested try-catch that was introduced in this PR. I think it can be refactored by moving part of the code to a separate function

kylewhite21 commented:
I did see the linter gripe (which yes was me). But this is failing in both common tests on commit https://github.com/seriyps/logger_journald/commit/34b5d9cc0c35b06abe05fe2b5b7d826bf69fd8bc

```
=== Ended at 2023-10-17 13:46:43
=== Location: [{prim_inet,split},
 {prim_inet,get_addr,0},
 {prim_inet,recvfrom0,0},
 {journald_server_mock,recv_raw,31},
 {journald_server_mock,recv_parse,35},
 {logger_journald_h_SUITE,just_log_case,71},
 {test_server,ts_tc,1783},
 {test_server,run_test_case_eval1,1292}]
=== === Reason: no function clause matching prim_inet:split(0,
                                                             <<"ERL_GROUP_LEADER=<0.350.0>\nERL_PID=<0.374.0>\nMESSAGE=test\nPRIORITY=6\nSYSLOG_TIMESTAMP=2023-10-17T13:46:43.906259-04:00\n">>,
                                                             []) 
  in function  prim_inet:get_addr/2 
  in call from prim_inet:recvfrom0/3 
  in call from journald_server_mock:recv_raw/1 (/Users/kwhite/vc/pe/logger_journald/test/journald_server_mock.erl, line 31)
  in call from journald_server_mock:recv_parse/1 (/Users/kwhite/vc/pe/logger_journald/test/journald_server_mock.erl, line 35)
  in call from logger_journald_h_SUITE:just_log_case/1 (/Users/kwhite/vc/pe/logger_journald/test/logger_journald_h_SUITE.erl, line 71)
  in call from test_server:ts_tc/3 (test_server.erl, line 1783)
  in call from test_server:run_test_case_eval1/6 (test_server.erl, line 1292)
```

seriyps commented:
CI fixed and I added a few more tests to test that feature

kylewhite21 commented:
Looks great 💪 thanks!
